compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -864,27 +864,8 @@ impl<'hir> LoweringContext<'_, 'hir> {
         fn is_ordinary(lower_ctx: &mut LoweringContext<'_, '_>, lhs: &Expr) -> bool {
             match &lhs.kind {
                 ExprKind::Array(..) | ExprKind::Struct(..) | ExprKind::Tup(..) => false,
-                ExprKind::Call(callee, ..) => {
-                    // Check for tuple struct constructor.
-                    if let ExprKind::Path(qself, path) = &callee.kind {
-                        let qpath = lower_ctx.lower_qpath(
-                            callee.id,
-                            qself,
-                            path,
-                            ParamMode::Optional,
-                            ImplTraitContext::disallowed(),
-                        );
-                        match qpath {
-                            hir::QPath::Resolved(
-                                _,
-                                hir::Path { res: Res::Def(DefKind::Ctor(..), _), .. },
-                            ) => false,
-                            _ => true,
-                        }
-                    } else {
-                        true
-                    }
-                }
+                // Check for tuple struct constructor.
+                ExprKind::Call(callee, ..) => !matches!(callee.kind, ExprKind::Path(..)),
```

Comment:
```suggestion
                ExprKind::Call(callee, ..) => !matches!(callee.kind, ExprKind::Path(None, _)),
```
If the goal is syntactic symmetry with patterns, then we need to reject `<A as B>::C(a, b)` here and in `destructure_assign` because such syntax is not supported in tuple struct patterns.

Reply:
This needs to be done for the nested case in `destructure_assign` as well.

Reply:
Done.